src/hotspot/cpu/aarch64/globals_aarch64.hpp

```diff
@@ -127,6 +127,8 @@ define_pd_global(intx, InlineSmallCode,          1000);
           range(1, 99)                                                  \
   product(ccstr, UseBranchProtection, "none",                           \
           "Branch Protection to use: none, standard, pac-ret")          \
+  product(bool, AlwaysMergeDMB, false,                                  \
```

Comment:
It should really be a diagnostic option if we're going to have it configurable at all, as a product option needs a CSR and it's not something end users would want to fiddle with.

Reply:
Exactly so, yes. I think we should simply stop merging DMBs where the end result of doing so is to strengthen the barrier beyond what was requested.